aea/helpers/transaction/base.py

```diff
@@ -235,10 +287,12 @@ def decode(cls, signed_transaction_protobuf_object) -> "SignedTransaction":
         :param signed_transaction_protobuf_object: the protocol buffer object whose type corresponds with this class.
         :return: A new instance of this class that matches the protocol buffer object in the 'signed_transaction_protobuf_object' argument.
         """
-        signed_transaction = pickle.loads(  # nosec
+        signed_transaction_dict = DictProtobufStructSerializer.decode(
             signed_transaction_protobuf_object.signed_transaction
         )
-        return signed_transaction
+        return cls(
+            signed_transaction_dict["ledger_id"], dict(signed_transaction_dict["body"])
```

Comment:
Shouldn't it be the responsibility of DictProtobufStructSerializer.decode to convert the nested objects to correct form (e.g. dict)?

Reply:
yes, that is correct, it is not needed, just forget to remove the conversion